hidden-posts.php

```diff
@@ -117,6 +121,21 @@ static function remove_post( $id ) {
         array_splice( $posts, array_search( $id, $posts ), 1 );
 
         update_option( self::META_KEY, array_map( 'intval', $posts ) );
+
+        // Update post meta.
+        update_post_meta( $id, self::META_KEY, 0 );
+    }
+
+    /**
+     * Migrate IDs of hidden posts by duplicating them from wp_options to wp_postmeta.
+     */
+    public function migrate_posts() {
```

Comment:
Does this need to run on every update/install? 
Is `$posts` always guaranteed to be an array? 
What about for sites that have no hidden posts?

Reply:
> Does this need to run on every update/install?
Given the current implementation, the update routine would run with every plugin update. I think it might be better to run the update routine based on the used version.

> Is $posts always guaranteed to be an array?
If $posts has at least one element, it's an `array`, otherwise it's `null`.

> What about for sites that have no hidden posts?
In this case, the migration should not be executed. I should have added a check for that.

Reply:
@mikeyarce Thanks to @htdat's comment in https://github.com/Automattic/hidden-posts/pull/22#pullrequestreview-615195672, I improved the update script in a way that it checks to the plugin version and only triggers the migration script when needed. 😀